projects/step-core/src/lib/services/dashboard.service.ts

```diff
@@ -4,42 +4,69 @@ import { AuthService } from './auth.service';
 import { ViewRegistryService } from './view-registry.service';
 import { downgradeInjectable, getAngularJSGlobal } from '@angular/upgrade/static';
 import { AJS_MODULE } from '../shared';
+import { lastValueFrom, Observable, ReplaySubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DashboardService {
-  private isGrafanaAvailable?: boolean = undefined;
+  private isGrafanaAvailable?: boolean;
+  private dashboardLink: ReplaySubject<any> = new ReplaySubject(1);
+  private requestingGrafanaAvailable = false;
 
   constructor(
     private _http: HttpClient,
     private _authService: AuthService,
     private _viewRegistryService: ViewRegistryService
   ) {}
 
-  getDashboardLink(taskId: string): string {
-    if (typeof this.isGrafanaAvailable === 'undefined') {
-      this.checkAvailability();
+  /*
+   * Removed this and use dashboardLink pipe (dashboard-link.pipe.ts) instead when migrating to NG2+
+   */
+  getDashboardLinkAJS(taskId: string): string {
+    if (!this.requestingGrafanaAvailable && typeof this.isGrafanaAvailable === 'undefined') {
+      this.checkAvailability(taskId);
     }
-    if (this.isGrafanaAvailable) {
+
+    return this.makeDashboardLink(taskId, this.isGrafanaAvailable);
+  }
+
+  getDashboardLink(taskId: string): Observable<string> {
+    if (!this.requestingGrafanaAvailable && typeof this.isGrafanaAvailable === 'undefined') {
+      this.checkAvailability(taskId);
+    } else {
+      this.dashboardLink.next(this.makeDashboardLink(taskId, this.isGrafanaAvailable));
+    }
+
+    return this.dashboardLink;
+  }
+
+  private makeDashboardLink(taskId: string, isGrafanaAvailable?: boolean) {
+    if (isGrafanaAvailable) {
       return '/#/root/grafana?d=3JB9j357k&orgId=1&var-taskId_current=' + taskId;
     } else {
       return '/#/root/dashboards/__pp__RTMDashboard?__filter1__=text,taskId,' + taskId;
     }
   }
 
-  checkAvailability(override = false) {
+  checkAvailability(taskId: string) {
```

Comment:
Since it is a void method, I would add a /** documentation on top of the method, to explain what it does behind. Will it raise an error, will it set some flags? etc..

Reply:
This method is only for the class itself so no documentation is needed. But you are right, we should mark it as private in this case.